analysis/bulk-deconvolution/exploratory-notebooks/compare-epic-quantiseq.Rmd

```diff
@@ -0,0 +1,441 @@
+# sum the quantiseq macrophage to be a single cell type
+quantiseq_macro <- deconv_df |> 
+  dplyr::filter(stringr::str_starts(celltype, "Macrophages.M")) |>
+  dplyr::group_by(project_name, project_id, sample_id, method) |>
+  dplyr::summarize(proportion = sum(proportion)) |>
+  dplyr::ungroup() |>
+  dplyr::mutate(celltype_consensus = "Macrophages")
+
+# update consensus df to have this new macrophage grouping for quantiseq
+consensus_macrophages_df <- consensus_df |>
+  dplyr::filter(
+    !stringr::str_starts(celltype_consensus, "Macrophages.M")
+  ) |>
+  dplyr::bind_rows(quantiseq_macro)
+```
+
+
+```{r, fig.width = 12, fig.height = 6, warning = FALSE}
+# Prepare some plotting colors
+tref_shared <- c("B cells", "CD4 T cells", "CD8 T cells", "NK cells", "Macrophages")
+bref_shared <- c("B cells", "CD4 T cells", "CD8 T cells", "NK cells", "Monocytes", "Neutrophils")
+other_celltypes_label <- "All other reference-specific cell types"
+other_colors <- c( 
+  "Other" = "grey95",
+  "All other reference-specific cell types" = "grey75"
+)
+
+tref_shared_color_map <- c(other_colors, color_map_tref[tref_shared])
+bref_shared_color_map <- c(other_colors, color_map_bref[bref_shared])
+ 
+
+# plot each project
+project_ids |> 
+  purrr::walk(
+    \(pid) {
+            
+      # determine which cell types to plot
+      if (pid %in% tref_projects) {
+        plot_celltypes <- tref_shared
+        color_map <- tref_shared_color_map
+      } else {
+        plot_celltypes <- bref_shared
+        color_map <- bref_shared_color_map
+      }
+  
+      # Subset to only this project and update celltypes to be:
+      # - In the shared vector
+      # - "Other"
+      # - not other but unshared
+      project_df <- consensus_macrophages_df |>
+        dplyr::filter(project_id == pid) |>
+        tidyr::drop_na() |>
+        dplyr::mutate(
+          celltype_grouped = ifelse(
+            celltype_consensus %in% c("Other", plot_celltypes), 
+            celltype_consensus,
+            other_celltypes_label
+          ),
+          celltype_grouped = forcats::fct_relevel(celltype_grouped, names(color_map)),
+          celltype_grouped = forcats::fct_rev(celltype_grouped)
+        ) 
+        
+      
+      print(make_barplot(project_df, color_map, cell_type_column = celltype_grouped))
+      
+    }
+  )
+```
+
+
+### Conclusions
+
+
+* Across all inferences from both methods, B cells are fairly uncommon, but `quanTIseq` generally estimates higher fractions compared to `EPIC`
+* Considering projects `SCPCP000001` and `SCPCP000002` (high grade and low grade gliomas, respectively):
+	* NK cells are commonly identified by `quanTIseq` but not by `EPIC`
+	* There appears to be some amount of consistency between macrophage inferences between methods
+	* `EPIC` estimates substantial portions of endothelial cells, which are not present in `quanTIseq`. Similarly, `quanTIseq` estimates substantial proportions of dendritic cells, which are not present in `EPIC`.
```

Comment:
Maybe also add a note about Monoctyes. It looks like a large portion of these samples are either Monoctyes, Macrohpages, or NK cells.

Reply:
Also just a side note that endothelial cells are not present in the top represented cell types using the consensus labels for this project, so that might be a reason at least for these projects to not use EPIC (at least with this reference).